### ISSUE TITLE
The training loss is always nan.

### ISSUE THREAD

**Issue by LuoXubo** [milesial/Pytorch-UNet#479]
Hi milesial, thanks for your nice work! However, when I was training the U-Net under the instruction of the README, the training loss is always "nan" and the validation dice score is a very small number, like 8.114e-12. Could you help me solve this problem? Thanks a lot!
![WeChat06f8244e19325314f8116b1cd45e4771](https://github.com/milesial/Pytorch-UNet/assets/53671855/c7c62dd6-ac2d-4129-8aa0-6c4bf54df305)

**Comment by binbin395** [comment]
I am getting the same issue as you got, but after switched the pytorch version from 2.1 back to 1.3, it seems OK now. Hope it helps.

**Comment by yuhanc0205** [comment]
I am experiencing the same issue, I have used all the default settings and the Carvana Dataset,  but my loss is always nan and dice score is not changing during training. Did you find any solution ?

**Comment by binbin395** [comment]
I found if revert it to the tag v4.0, it's ok, maybe some one can find which commit after that version involved the problem.

**Comment by benlin1211** [comment]
I managed to solve this problem by turning off mixed precision flag.
That is, instead of using `python train.py --amp`, 
**use `python train.py` to train the code.**
Although it takes more time and memory during training, the code can be trained successfully.
 
Similar issue: https://github.com/pytorch/pytorch/issues/40497

**Comment by erikbwu** [comment]
Thank you @benlin1211!!
